ngraph_bridge/ngraph_find_replace_prefetchdataset.h

```diff
@@ -0,0 +1,140 @@
+/*******************************************************************************
+ * Copyright 2017-2019 Intel Corporation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *******************************************************************************/
+
+#include "tensorflow/core/graph/graph.h"
+#include "tensorflow/core/graph/node_builder.h"
+
+#include "ngraph_bridge/ngraph_api.h"
+#include "ngraph_bridge/ngraph_prefetch_shared_data.h"
+#include "ngraph_bridge/ngraph_utils.h"
+
+using namespace std;
+
+namespace tensorflow {
+
+namespace ngraph_bridge {
+
+Node* FindPrefetch(Node* makeiterator_node) {
+  Node* prefetch_node = nullptr;
+  for (auto e1 : makeiterator_node->in_edges()) {
+    Node* n = e1->src();
+    // resnet case
+    if (n->type_string() == "PrefetchDataset") {
+      prefetch_node = n;
+      break;
+    } else if (n->type_string() == "ModelDataset") {  // axpy case
+      for (auto e2 : n->in_edges()) {
+        if (e2->src()->type_string() == "OptimizeDataset") {
+          for (auto e3 : e2->src()->in_edges()) {
+            if (e3->src()->type_string() == "PrefetchDataset") {
+              prefetch_node = e3->src();
+              break;
+            }
+          }
+        } else {
+          // this is neither axpy nor resnet
+          NGRAPH_VLOG(0)
```

Comment:
VLOG(0) might not be required since we are throwing an error. and you can break in the else